server/models/procedures/voucher.sql

```diff
@@ -220,4 +220,80 @@ BEGIN
   END IF;
 END $$
 
+
+-- this is handled on change rate changed
+DROP PROCEDURE IF EXISTS `UpdateLossAccount`$$
+CREATE   PROCEDURE `UpdateLossAccount`(
+  IN _enterpriseId INT,
+  IN _project_id INT, 
+  IN _user_id INT,
+  IN _currencyId INT,
+  IN _date DATE, 
+  IN _rate DECIMAL(19, 4)
+  )
+BEGIN
+DECLARE _account_id, _loss_account_id, _gain_account_id INT(10) UNSIGNED;
+DECLARE _privousRate, _balance, _difference DECIMAL(19, 4);
```

Comment:
Could you adopt one style of variable name : either `camelCalse` or `underscore_case`.
If you choose `camelCase`, all your variables must be like : `_accountId`, `_lossAccountId`, `_gainAccountId`, `_previousRate`, etc.
If you choose `underscore_case` all your variables must be like : `_account_id`, `_loss_account_id`, `_gain_account_id`, `_previous_rate`, etc.